scripts/tests/test_zkai_one_transformer_block_surface_gate.py

```
@@ -0,0 +1,190 @@
+        fusion["route_matrix"]["fused_savings_bytes_total"] = 0
+        with self.assertRaisesRegex(gate.OneTransformerBlockSurfaceError, "fusion metrics must be positive"):
+            gate._component_rows(fusion, d64, d128, matrix)
+
+        fusion = gate.load_json(gate.FUSION_MECHANISM)
+        d128["summary"]["mutations_rejected"] = d128["summary"]["mutation_cases"] - 1
+        with self.assertRaisesRegex(gate.OneTransformerBlockSurfaceError, "d128 mutation rejection count drift"):
+            gate._component_rows(fusion, d64, d128, matrix)
+
+        d128 = gate.load_json(gate.D128_BLOCK_RECEIPT)
+        matrix["external_rows"][0]["proof_size_reported"] = "1 byte"
+        with self.assertRaisesRegex(gate.OneTransformerBlockSurfaceError, "NANOZK row drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)